pkg/cmd/pipeline/list.go

```diff
@@ -79,44 +99,50 @@ func listCommand(p cli.Params) *cobra.Command {
 				Out: cmd.OutOrStdout(),
 				Err: cmd.OutOrStderr(),
 			}
-			return printPipelineDetails(stream, p)
+			return printPipelineDetails(stream, p, opts.AllNamespaces, opts.NoHeaders)
 		},
 	}
 	f.AddFlags(c)
+	c.Flags().BoolVarP(&opts.AllNamespaces, "all-namespaces", "A", opts.AllNamespaces, "list pipelines from all namespaces")
+	c.Flags().BoolVarP(&opts.NoHeaders, "no-headers", "", opts.NoHeaders, "do not print column headers with output (default print column headers with output)")
 
 	return c
 }
 
-func printPipelineDetails(s *cli.Stream, p cli.Params) error {
+func printPipelineDetails(s *cli.Stream, p cli.Params, allnamespaces bool, noheaders bool) error {
 
 	cs, err := p.Clients()
 	if err != nil {
 		return err
 	}
 
-	ps, prs, err := listPipelineDetails(cs, p.Namespace())
+	ns := p.Namespace()
```

Comment:
So this still isn't working for `--all-namespaces`. As mentioned previously, my hunch is that you need to not use `p.Namespace`. You should try the approach you used with taskruns/pipelineruns, which was as follows:

```go
ns := p.Namespace()
if allnamespaces {
    ns = ""
}
```